Obsidian/Commands/MainCommandModule.cs

```diff
@@ -49,29 +50,36 @@ public async Task PluginsAsync()
             };
 
             var messages = new List<ChatMessage>();
-
+            int plugins_i = 0;
             foreach (var pluginContainer in Context.Server.PluginManager.Plugins)
+            {
                 if (pluginContainer.Info.ProjectUrl != null)
                 {
                     messages.Add(new ChatMessage
                     {
-                        Text = ChatColor.DarkGreen + pluginContainer.Info.Name + $"{ChatColor.Reset}, ",
+                        Text = ChatColor.DarkGreen + pluginContainer.Info.Name,
                         ClickEvent = new TextComponent { Action = ETextAction.OpenUrl, Value = pluginContainer.Info.ProjectUrl.AbsoluteUri }
                     });
                 }
                 else
                 {
                     messages.Add(new ChatMessage
                     {
-                        Text = ChatColor.DarkGreen + pluginContainer.Info.Name + $"{ChatColor.Reset}, "
+                        Text = ChatColor.DarkGreen + pluginContainer.Info.Name
                     });
                 }
```

Comment:
Try using a variable for the ClickEvent property, that gets set when there's a ProjectUrl, so that you can reduce duplicate Add(...) calls. (ask me for example code if you want to)

Reply:
```suggestion
                var plugin = new ChatMessage();
                plugin.Text = ChatColor.DarkGreen + pluginContainer.Info.Name;
                if (pluginContainer.Info.ProjectUrl != null) plugin.ClickEvent = new TextComponent { Action = ETextAction.OpenUrl, Value = pluginContainer.Info.ProjectUrl.AbsoluteUri };

                messages.Add(new ChatMessage
                {
                    Text = $"{ChatColor.Reset}{(i + 1 < Context.Server.PluginManager.Plugins.Count ? ", " : "")}"
                });
```
like this? Or I misunderstood you?

Reply:
Yep, that's what I wanted.

Reply:
Ok. I'll add this as well.